Week-13/index.js

```diff
@@ -0,0 +1,101 @@
+const express = require("express");
+const request = require("request");
+
+const app = express();
+
+//requesting data for a particular city
+app.get("/", (req, res) => {
+  let city = req.query.city;
+  var request = require("request");
+
+  request(
+    `http://api.weatherapi.com/v1/current.json?key=372f208a3f284a61b4670325222910&q=${city}&aqi=no`,
+    function (error, response, body) {
+      if (response.statusCode === 200) {
+        res.send(`The weather in your city ${city} is ${body}`);
+      }
+    }
+  );
+});
+//requesting data for multiple cities
+app.get("/", (req, res) => {
+  let cities = req.query.cities;
+  var request = require("request");
+
+  const getWeatherData = () =>
+    Promise.all(
+      cities.map(
+        (city) =>
+          new Promise((resolve, reject) =>
+            request.get(
+              `http://api.weatherapi.com/v1/current.json?key=372f208a3f284a61b4670325222910&q=${city}&aqi=no`,
+              function (error, response, body) {
+                if (response.statusCode === 200) {
+                  return resolve(body);
+                }
+                return reject(error);
+              }
+            )
+          )
+      )
+    );
+
+  const main = async () => {
+    const data = await getWeatherData();
+    res.send(data);
+  };
+  main();
+});
+//requesting data by pincode
+app.get("/zipcode", (req, res) => {
+  let postalcode = req.query.code;
```

Comment:
instead of writing the handler inline, it is a good coding practice to provide a handler and/or group similar or related handler to a particular file. Also, here you are using weather api in all the functions. It is a good coding practice to abstract the weather api methods to a service class and then use them here as class methods.